ui/components/app/transaction-activity-empty-state/transaction-activity-empty-state.tsx

```diff
@@ -1,35 +1,65 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { twMerge } from '@metamask/design-system-react';
+import { InternalAccount } from '@metamask/keyring-internal-api';
+import { EthMethod } from '@metamask/keyring-api';
 import { TabEmptyState } from '../../ui/tab-empty-state';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import { useTheme } from '../../../hooks/useTheme';
+import { getUseExternalServices, getIsSwapsChain } from '../../../selectors';
+import { getCurrentChainId } from '../../../../shared/modules/selectors/networks';
+import { MetaMetricsSwapsEventSource } from '../../../../shared/constants/metametrics';
+import useBridging from '../../../hooks/bridge/useBridging';
 
 export type TransactionActivityEmptyStateProps = {
   /**
    * Additional className to apply to the component
    */
   className?: string;
+  /**
+   * The account to use for swap logic
+   */
+  account: InternalAccount;
 };
 
 export const TransactionActivityEmptyState: React.FC<
   TransactionActivityEmptyStateProps
-> = ({ className }) => {
+> = ({ className, account }) => {
   const t = useI18nContext();
   const theme = useTheme();
+  const isSigningEnabled =
+    account.methods.includes(EthMethod.SignTransaction) ||
+    account.methods.includes(EthMethod.SignUserOperation);
+  const isExternalServicesEnabled = useSelector(getUseExternalServices);
+  const chainId = useSelector(getCurrentChainId);
+  const isSwapsChain = useSelector((state) => getIsSwapsChain(state, chainId));
+
+  const { openBridgeExperience } = useBridging();
 
   // Theme-aware icon selection
   const activityIcon =
     theme === 'dark'
       ? './images/empty-state-activity-dark.png'
       : './images/empty-state-activity-light.png';
 
+  const handleSwapOnClick = useCallback(async () => {
+    openBridgeExperience(MetaMetricsSwapsEventSource.MainView, undefined, true);
+  }, [openBridgeExperience]);
+
   return (
     <TabEmptyState
       icon={
         <img src={activityIcon} alt={t('activity')} width={72} height={72} />
       }
       description={t('activityEmptyDescription')}
-      className={className}
+      actionButtonText={t('swapTokens')}
+      actionButtonProps={{
+        isDisabled:
+          !isSwapsChain || !isSigningEnabled || !isExternalServicesEnabled,
```

Comment:
One thing we may want to check is whether we can still use this component via the Snaps SDK / Snaps API now that the type has changed from a string to an enum. Since the Snap UI is accessed through the Snaps API, we may need to continue supporting string types. Let's confirm with the @MetaMask/snaps team